src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -56,6 +57,23 @@ public void PopulateFeature(IEnumerable<ApplicationPart> parts, ViewsFeature fea
         /// <returns>The <see cref="ViewInfoContainer"/> <see cref="Type"/>.</returns>
         protected virtual Type GetViewInfoContainerType(AssemblyPart assemblyPart)
         {
+#if NETSTANDARD1_6
+            if (assemblyPart.Assembly.Location != null)
+            {
+                var precompiledAssemblyFileName = assemblyPart.Assembly.GetName().Name
+                    + PrecompiledViewsAssemblySuffix
+                    + ".dll";
+                var precompiledAssemblyFilePath = Path.Combine(
+                    assemblyPart.Assembly.Location,
```

Comment:
We need to special case this being null. `Path.Combine` will collapse null so this would be equivalent to `.\Foo.Views.dll`